deploy/cluster_role.yaml

```diff
@@ -0,0 +1,54 @@
+## this is an extra set of permissions that the Jaeger Operator might make use of if granted
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: jaeger-operator-cluster
```

Comment:
nit the suffix `-cluster` seems redundant.

Reply:
Do you have a suggestion?

Reply:
Skip the suffix?

Reply:
Clashes with the regular role:

https://github.com/jaegertracing/jaeger-operator/blob/86a656fae219ce84406fec1c06fa01b0d86fc3c5/deploy/role.yaml#L2-L6

Reply:
It's a different object it should not matter.

Reply:
It's a different object type it should not matter.

Reply:
As an admin, I'm not sure I would be happy to see two objects with the same name (`jaeger-operator`) across very similar objects (`ClusterRole` and `Role`). When appearing in logs and error messages, it's clear that `jaeger-operator-cluster` is about the cluster role, and not role.